moveit_py/src/moveit/moveit_core/robot_state/robot_state.cpp

```diff
@@ -0,0 +1,507 @@
+          Representation of a robot's state.
+          At the lowest level, a state is a collection of variables. Each variable has a name and can have position, velocity, acceleration and effort associated to it. Effort and acceleration share the memory area for efficiency reasons (one should not set both acceleration and effort in the same state and expect things to work). Often variables correspond to joint names as well (joints with one degree of freedom have one variable), but joints with multiple degrees of freedom have more variables. Operations are allowed at variable level, joint level (see JointModel) and joint group level (see JointModelGroup).
+                  For efficiency reasons a state computes forward kinematics in a lazy fashion. This can sometimes lead to problems if the update() function was not called on the state.
+          )")
+
+      .def(py::init<const std::shared_ptr<const moveit::core::RobotModel>&>(),
+           R"(
+           Initializes robot state from a robot model.
+
+           Args:
+               :py:class:`moveit_py.core.RobotModel`: The robot model associated to the instantiated robot state.
+
+           )")
+
+      // Get underlying robot model, frame transformations and jacobian
+      .def_property("robot_model", &moveit::core::RobotState::getRobotModel, nullptr,
+                    py::return_value_policy::reference,
+                    R"(
+                    :py:class:`moveit_py.core.RobotModel`: The robot model instance associated to this robot state.
+                    )")
+
+      .def_property("dirty", &moveit::core::RobotState::dirty, nullptr,
+                    R"(
+            bool: True if the robot state is dirty.
+            )")
+
+      .def("get_frame_transform", &moveit_py::bind_robot_state::get_frame_transform, py::arg("frame_id"),
+           py::return_value_policy::move,
+           R"(
+           Get the transformation matrix from the model frame (root of model) to the frame identified by frame_id.
+           If frame_id was not found, frame_found is set to false and an identity transform is returned.
+       This method is restricted to frames defined within the robot state and doesn't include collision object present in the collision world. Please use the PlanningScene.get_frame_transform method for collision world objects.
+           Args:
+               frame_id (str): The id of the frame to get the transform for.
+           Returns:
+               :py:class:`numpy.ndarray`: The transformation matrix from the model frame to the frame identified by frame_id.
+           )")
+
+      .def("get_pose", &moveit_py::bind_robot_state::get_pose, py::arg("link_name"),
+           R"(
+           Get the pose of a link that is defined in the robot model.
+           Args:
+               link_name (str): The name of the link to get the pose for.
+           Returns:
+               geometry_msgs.msg.Pose: A ROS geometry message containing the pose of the link.
```

Comment:
Missing the `:py:class` bit here

Reply:
[348f9b8](https://github.com/ros-planning/moveit2/pull/1546/commits/348f9b88b99927a9217882749671bcb6a47504a9)